panel/widgets/tables.py

```diff
@@ -197,6 +161,493 @@ def _get_columns(self):
+                    model.groupings = self._get_groupings()
+            elif hasattr(self, '_update_' + event.name):
+                getattr(self, '_update_' + event.name)(model)
+            else:
+                self._update_columns(event, model)
+
+    def _filter_dataframe(self, df, ):
+        """
+        Filter the DataFrame.
+
+        Parameters
+        ----------
+        df : DataFrame
+           The DataFrame to filter
+        query : dict
+            A dictionary containing all the query parameters
+
+        Returns
+        -------
+        DataFrame
+            The filtered DataFrame
+        """
+        filters = []
+        for col_name, filt in self._filters:
+            if isinstance(filt, FunctionType):
+                df = filt(df)
+                continue
+            val = getattr(filt.owner, filt.name)
+            column = df[col_name]
+            if np.isscalar(val):
+                mask = column == val
+            elif isinstance(val, list):
+                if not val:
+                    continue
+                mask = column.isin(val)
+            elif isinstance(val, tuple):
+                start, end = val
+                if start is None and end is None:
+                    continue
+                elif start is None:
+                    mask = column<=end
+                elif end is None:
+                    mask = column>=start
+                else:
+                    mask = (column>=start) & (column<=end)
+            else:
+                raise ValueError(f"'{col_name} filter value not "
+                                 "understood. Must be either a scalar, "
+                                 "tuple or list.")
+            filters.append(mask)
+        if filters:
+            mask = filters[0]
+            for f in filters:
+                mask &= f
+            df = df[mask]
+        return df
+
+    def add_filter(self, filter, column=None):
+        """
+        Adds a filter to the table which will trigger updates when changed.
+
+        When a widget or parameter is supplied the filtering will follow
+        a few well defined behvaiors:
+
+            - scalar: Filters by checking for equality
+            - tuple: A tuple will be interpreted as range.
+            - list: A list will be interpreted as a set of discrete
+                    scalars and the filter will check if the values
+                    in the column match any of the items in the list.
+
+        Arguments
+        ---------
+        filter: Widget, param.Parameter or FunctionType
+            The filter which will provide the value to filter the
+            DataFrame along the declared column or a function which
+            will be passed the DataFrame and should return a filtered
+            copy of the DataFrame.
+        column: str or None
+            If the filter is a widget or parameter the column declares
+            the value will filter on.
```

Comment:
```suggestion
            Column to which the filter will be applied, if the filter is a widget or parameter.
```

(Is that what it meant?)